namespace cpp infra 
namespace java infra.gen

const string DATASPHERES_ROOT		        	= "dataspheres"
const string NODES_ROOT 		            	= "nodes"
const string SERVICES_ROOT		            	= "services"
const string DATOM_ROOT                             	= "/datom"
const string CONFIGSERVICE_ROOT                     	= "/configservice"
const string DATASPHERE_ROOT_PATH_FORMAT            	= "/dataspheres/{{}}"
const string SERVICES_ROOT_PATH_FORMAT              	= "/dataspheres/{{}}/services"
const string SERVICE_ROOT_PATH_FORMAT               	= "/dataspheres/{{}}/services/{{}}"


const string PB_SPHERE_ROOT_PATH_FORMAT			= "/dataspheres/{{}}/pbsphere/{{}}"
const string PB_SPHERE_RINGS_ROOT_PATH_FORMAT		= "/dataspheres/{{}}/pbsphere/{{}}/rings"
const string PB_SPHERE_RING_ROOT_PATH_FORMAT		= "/dataspheres/{{}}/pbsphere/{{}}/rings/{{}}"
const string PB_SPHERE_RESOURCES_ROOT_PATH_FORMAT	= "/dataspheres/{{}}/pbsphere/{{}}/resources"
const string PB_SPHERE_RESOURCE_ROOT_PATH_FORMAT	= "/dataspheres/{{}}/pbsphere/{{}}/resources/{{}}"

const string TOPIC_SERVICES                         	= "services"
const string TOPIC_PB_SPHERE_RESOURCES			= "pbsphere.{{}}"
